main/src/library/Regex.flix

```diff
@@ -121,91 +104,90 @@ mod Regex {
 
 
     ///
-    /// Compile the regular expression `regex` into a Pattern.
+    /// Compile the regular expression `regex` into a Regex object (internally
+    /// a Regex is `java.util.regex.Pattern`).
     ///
     /// Returns `Err` if the regex is ill-formed and cannot be compiled.
     ///
     /// The syntax of regular expressions is the same as Java, see the JDK Javadocs
     /// of the class `util.regex.Pattern` for reference.
     ///
-    pub def compile(regex: String): Result[String, Pattern] =
+    pub def compile(regex: String): Result[String, Regex] =
```

Comment:
Two conflicting thoughts:

- If we drop `compile` then we have guarantee that regexs are always valid (because they must be made with `Regex#` which is checked at compile-time. This avoids all kinds of injection attacks. I know in JS-world this is a big problem.
- If we don't drop it, should we perhaps have `String.toRegex` or similar?

Reply:
We could also name it `compileUnsafeRegex` (or whatever).

Reply:
In any case, the documentation should probably strongly suggest not to use this method, but to write `Regex#` instead.

Reply:
I think we definitely want to a `compileWithFlags` whether or not we call it that.

I like the idea of `String.toRegex` as it fits nicely with how Flix's favours `toX` functions - maybe have both `String.toRegex` and String.toRegexWithFlags` ?

Reply:
> I like the idea of String.toRegex as it fits nicely with how Flix's favours toX functions - maybe have both String.toRegex and String.toRegexWithFlags` ?

Yes

Reply:
Moved into String.